go.go

```diff
@@ -687,6 +781,54 @@ const goUnmarshalField = `{{if eq .Type "bool"}}
 		i++
 	}
 {{else if eq .Type "int32"}}
+{{- if .TypeList}}
+	if header == {{.Index}} {
+	{{template "unmarshal-varint" .}}
+		if x > uint(ColferListMax) {
+			return 0, ColferMax(fmt.Sprintf("colfer: {{.String}} length %d exceeds %d elements", x, ColferListMax))
+		}
+
+		l := int(x)
+
+		if end := i + l*4; end >= len(data) {
```

Comment:
Doesn't work here with variable length. This check can be removed, because the bounds are correctly validated in the loop below.

Reply:
Dito for the 64-bit version below.

Reply:
changed